endpoints/decorators/metrics_test.go

```diff
@@ -51,6 +52,58 @@ func TestBadPutRequestMetrics(t *testing.T) {
 	assert.Equal(t, metricstest.MockHistograms["puts.current_url.duration"], 0.00, "Unsuccessful put request duration should have been logged")
 }
 
+func TestCustomKeyPutRequestMetrics(t *testing.T) {
+	metrics := metricstest.CreateMockMetrics()
+
+	type testExpectedValues struct {
+		totalRequests     int64
+		badRequests       int64
+		customKeyRequests int64
+	}
+	testCases := []struct {
+		desc                  string
+		inHandler             func(w http.ResponseWriter, _ *http.Request, _ httprouter.Params)
+		expectedCounterValues testExpectedValues
+	}{
+		{
+			desc: "A put request that comes with its own custom key and Put endpoint throws no error",
+			inHandler: func(w http.ResponseWriter, _ *http.Request, _ httprouter.Params) {
+				w.WriteHeader(CacheUpdateCode)
+				w.WriteHeader(200)
+			},
+			expectedCounterValues: testExpectedValues{
+				totalRequests:     int64(1),
+				badRequests:       int64(0),
+				customKeyRequests: int64(1),
+			},
+		},
+		{
+			desc: "A put request that comes with its own custom key and Put endpoint throws error",
+			inHandler: func(w http.ResponseWriter, _ *http.Request, _ httprouter.Params) {
+				w.WriteHeader(CacheUpdateCode)
+				w.WriteHeader(400)
+			},
+			expectedCounterValues: testExpectedValues{
```

Comment:
Seems that you have values of "2" here because this is the second test. It may be better to clear the metrics between tests. Would make the test cleaner, and easier to add cases. I did spend some time wondering why the first set of two `w.WriteHeader()` calls was only 1 request, while the second set resulted in 2 requests.

Reply:
You are right, sorry about that. Modified to clear the metrics at every test case.